### PR TITLE
chore(deps): bump enr, discv5, secp256k1

### PR DESCRIPTION
**Pull request by DaniPopes**
- `KeyPair` -> `Keypair`
- `Message::from_slice(&[u8])?` -> `Message::from_digest([u8; 32])`

~~Blocked on `enr` updating their `secp256k1` https://github.com/sigp/enr/pull/61.~~

~~Blocked on a `discv5` release: https://github.com/sigp/discv5/issues/244~~

### CODE REVIEW

File: crates/interfaces/src/test_utils/generators.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@ use reth_primitives::{
     SealedHeader, StorageEntry, Transaction, TransactionKind, TransactionSigned, TxLegacy, B256,
     U256,
 };
-use secp256k1::{KeyPair, Secp256k1};
+use secp256k1::{Keypair, Secp256k1};
```
Comment (mattsse) [inline crates/interfaces/src/test_utils/generators.rs:10]: lol why

**Review by mattsse — APPROVED** [review]
more breaking changes 👍

lgtm, when it compiles

File: crates/interfaces/src/test_utils/generators.rs (hunk 1 above)
Comment (DaniPopes) [reply crates/interfaces/src/test_utils/generators.rs:10]: Apparently mistake in the Rust library https://github.com/rust-bitcoin/rust-secp256k1/issues/647

**Comment by DaniPopes** [comment]
Blocked on @mattsse PR :smile: https://github.com/sigp/enr/pull/61

**Comment by Rjected** [comment]
> Blocked on @mattsse PR 😄 [sigp/enr#61](https://github.com/sigp/enr/pull/61)

cc @mattsse this is merged!

**Comment by DaniPopes** [comment]
Looks like `enr 0.10.1` was released with breaking changes, and `discv5` will also need to be updated.

**Comment by citizen-stig** [comment]
Looking forward for this PR to be merged. 

I am probably missing something, but `discv5` in Cargo.toml is 0.4 and in Cargo.lock is 0.4.1 , which is latest in crates.io as for today.

Also, I want to point out, that currently there are 2 version of secp256k1 in dependency tree: `revm-primitives` brings 0.28.2 already. So merging this could slightly improve compilation time.

**Comment by DaniPopes** [comment]
Yes, that's correct. But discv5 has not yet updated to using alloy-rlp so until that is done this PR cannot be merged.

**Comment by citizen-stig** [comment]
@DaniPopes thank you for clarification!

**Comment by DaniPopes** [comment]
https://github.com/sigp/enr/issues/69

**Comment by emhane** [comment]
discv5 is configured to a git commit, and should remain so. for `reth_discv5` to work we need at least commit 3c0d3d7 on https://github.com/sigp/discv5/commits/master/, but ideally we want commit 15b1725 to be able to update the local enr with fork id after starting the node which isn't possible otherwise without some hacks.

File: Cargo.toml [hunk 2]
```diff
@@ -320,9 +318,10 @@ futures-util = "0.3.25"
 hyper = "0.14.25"
 tower = "0.4"
 tower-http = "0.4"
+reqwest = "0.12"
 
 # p2p
-discv5 = { git = "https://github.com/sigp/discv5", rev = "04ac004" }
+discv5 = "0.4.1"
```
Comment (emhane) [inline Cargo.toml:324]:
```suggestion
discv5 = { git = "https://github.com/sigp/discv5", rev = "15b1725" }
```

File: Cargo.toml [hunk 3]
```diff
@@ -331,11 +330,11 @@ jsonrpsee-core = "0.20"
 jsonrpsee-types = "0.20"
 
 # crypto
-secp256k1 = { version = "0.27.0", default-features = false, features = [
+secp256k1 = { version = "0.28", default-features = false, features = [
     "global-context",
     "recovery",
 ] }
-enr = { version = "=0.10.0", default-features = false, features = ["k256"] }
+enr = { version = "0.11", default-features = false, features = ["k256"] }
```
Comment (emhane) [inline Cargo.toml:337]:
```suggestion
enr = { version = "0.11", default-features = false, features = ["rust-secp256k1"] }
```

File: crates/net/dns/src/lib.rs [hunk 4]
```diff
@@ -402,8 +402,8 @@ fn convert_enr_node_record(enr: &Enr<SecretKey>) -> Option<DnsNodeRecordUpdate>
     }
     .into_ipv4_mapped();
 
-    let mut maybe_fork_id = enr.get(b"eth")?;
-    let fork_id = ForkId::decode(&mut maybe_fork_id).ok();
+    let maybe_fork_id = enr.get(b"eth")?;
+    let fork_id = ForkId::decode(&mut &maybe_fork_id[..]).ok();
```
Comment (emhane) [inline crates/net/dns/src/lib.rs:406]: nice

File: crates/net/discv5/Cargo.toml [hunk 5]
```diff
@@ -18,7 +18,6 @@ reth-metrics.workspace = true
 
 # ethereum
 alloy-rlp.workspace = true
-rlp = "0.5.2"
 discv5 = { workspace = true, features = ["libp2p"] }
 enr = { workspace = true, default-features = false, features = ["rust-secp256k1"] }
```
Comment (emhane) [inline crates/net/discv5/Cargo.toml:22]:
```suggestion
enr.workspace = true
```

**Review by emhane — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 2 above)
Comment (DaniPopes) [reply Cargo.toml]: 0.5.0 was released

File: Cargo.toml (hunk 3 above)
Comment (DaniPopes) [reply Cargo.toml]: looks like both are needed since without k256 it fails to compile: https://github.com/sigp/enr/pull/74

**Review by emhane — APPROVED** [review]

**Comment by mattsse** [comment]
I still think enr decoding is currently broken, resulting in lots of invalid signature errors on decoding

**Comment by emhane** [comment]
> I still think enr decoding is currently broken, resulting in lots of invalid signature errors on decoding

yes, blocked by <https://github.com/sigp/enr/pull/75>

**Comment by emhane** [comment]
discv5 bumped https://github.com/sigp/discv5/commit/e9e0d4f93ec35591832a9a8d937b4161127da87b
enr bumped https://github.com/sigp/enr/releases/tag/v0.12.0

**Comment by DaniPopes** [comment]
discv5 is still on `enr 0.11`

**Comment by emhane** [comment]
> discv5 is still on `enr 0.11`

enr bumped in discv5 https://github.com/sigp/discv5/releases/tag/v0.6.0

**Pull request merged**